v23.2.10 Cloud-first release notes (#18864)
cockroachdb/docs: src/current/_includes/releases/v23.2/v23.2.10.md

	- [`CREATE CHANGEFEED`]({% link v23.2/create-changefeed.md %})
	- [`CREATE EXTERNAL CONNECTION`]({% link v23.2/create-external-connection.md %})
	- [`CREATE SCHEDULE FOR BACKUP`]({% link v23.2/create-schedule-for-backup.md %})
	- [`CREATE SCHEDULE FOR CHANGEFEED`]({% link v23.2/create-schedule-for-changefeed.md %})
	- [`EXPORT`]({% link v23.2/export.md %})
	- [`IMPORT INTO`]({% link v23.2/import-into.md %})
	- [`RESTORE`]({% link v23.2/restore.md %})
	- [`SHOW BACKUPS`]({% link v23.2/show-backup.md %})
	- [`SHOW BACKUP`]({% link v23.2/show-backup.md %})

<h3 id="v23-2-10-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

- Added a new [Kafka sink]({% link v23.2/changefeed-sinks.md %}#kafka) utilizing the `franz-go` library and our own `batching_sink` behind a [cluster setting]({% link v23.2/cluster-settings.md %}) (`changefeed.new_kafka_sink_enabled`, disabled by default). [#128048][#128048]
- The v2 Kafka and Pub/Sub [changefeed sinks]({% link v23.2/changefeed-sinks.md %}) now display notices indicating the topics they will emit to. [#128459][#128459]

<h3 id="v23-2-10-sql-language-changes">SQL language changes</h3>

> - Added the `sql.auth.grant_option_for_owner.enabled` [cluster setting]({% link v23.2/cluster-settings.md %}). The default value is `true`, which results in behavior that matches the existing behavior of CockroachDB. When set to `false`, then the `GRANT OPTION` is not implcitly given to the owner of an object. The object owner still implicitly has all privileges on the object, just not the ability to grant them to other users. [#126958][#126958]

Reviewer: ```suggestion
- Added the [`sql.auth.grant_option_for_owner.enabled` cluster setting]({% link v23.2/cluster-settings.md %}#setting-sql-auth-grant-option-for-owner-enabled). The default value is `true`, which results in behavior that matches the existing behavior of CockroachDB. When set to `false`, then the `GRANT OPTION` is not implcitly given to the owner of an object. The object owner still implicitly has all privileges on the object, just not the ability to grant them to other users. [#126958][#126958]
```

- Fixed a bug where the `DISCARD` statement was disallowed when the [`default_transaction_read_only`]({% link v23.2/session-variables.md %}#default-transaction-read-only) session setting was set to `on`. [#127548][#127548]

<h3 id="v23-2-10-db-console-changes">DB Console changes</h3>

- The [**Databases** and **Tables** pages]({% link v23.2/ui-databases-page.md %}) in the DB Console now show a loading state while loading information for databases and tables, including size and range counts. [#127709][#127709]
- On the [**Databases** page]({% link v23.2/ui-databases-page.md %}) in the DB Console, table names will no longer appear with quotes around the schema and table name. [#127765][#127765]

<h3 id="v23-2-10-bug-fixes">Bug fixes</h3>

- Fixed a bug causing gateway nodes to crash while executing [`INSERT`]({% link v23.2/insert.md %}) statements in [`REGIONAL BY ROW`]({% link v23.2/table-localities.md %}#regional-by-row-tables) tables. This bug had been present since v23.2. [#127276][#127276]

> - Fixed a bug where dropping [`ENUM`]({% link v23.2/enum.md %}) values that were referenced by [index expressions]({% link v23.2/expression-indexes.md %}) could fail with an error. [#127453][#127453]

Reviewer: The enum page did not have info about how to drop a value:

```suggestion
- Fixed a bug where [dropping `ENUM` values]({% link v23.2/alter-type.md %}#drop-a-value-in-a-user-defined-type) that were referenced by [index expressions]({% link v23.2/expression-indexes.md %}) could fail with an error. [#127453][#127453]
```

- Fixed a bug that caused a memory leak when executing SQL statements with comments, e.g., [`SELECT /* comment */ 1;`]({% link v23.2/select-clause.md %}). Memory owned by a SQL session would continue to grow as these types of statements were executed. The memory would only be released when closing the SQL session. This bug had been present since v23.1. [#127758][#127758]
- Fixed a memory leak that could occur when specifying a non-existent [virtual cluster]({% link v23.2/cluster-virtualization-overview.md %}) name in the connection string. [#128104][#128104]
- Fixed a bug where [`CREATE INDEX IF NOT EXISTS`]({% link v23.2/create-index.md %}) would not correctly short-circuit if the given index already existed. [#128312][#128312]

> - Fixed a bug in overly eagier syntax validation, in which the `DESCENDING` clause was not allowed for non-terminal columns of an [inverted index]({% link v23.2/inverted-indexes.md %}). Only the last column of an inverted index should be prevented from being `DESCENDING`, and this is now properly checked. [#128312][#128312]

Reviewer: fix typo, and clarification suggestion:
```suggestion
- Fixed a bug in overly eager syntax validation, which did not allow the `DESCENDING` clause for non-terminal columns of an [inverted index]({% link v23.2/inverted-indexes.md %}). Only the last column of an inverted index should be prevented from being `DESCENDING`, and this is now properly checked. [#128312][#128312]
```

- Fixed a bug where an [index]({% link v23.2/indexes.md %}) could store a column in the primary index if that column had a mixed-case name. [#128312][#128312]
- Fixed small memory leaks that would occur during [changefeed creation]({% link v23.2/create-changefeed.md %}). [#128048][#128048]

> - Setting or dropping a default value on a [computed column]({% link v23.2/computed-columns.md %}) is now blocked -- even for null defaults. Previously, setting or dropping a default value on a computed column was a no-op. [#128467][#128467]

Reviewer: ```suggestion
- Setting or dropping a default value on a [computed column]({% link v23.2/computed-columns.md %}) is now blocked -- even for null defaults. Previously, setting or dropping a default value on a computed column was a no-op, now there will be an error message. [#128467][#128467]
```

- Fixed a bug that could cause spurious user permission errors when multiple databases shared a common schema with a routine referencing a table. The bug had existed since [user-defined functions]({% link v23.2/user-defined-functions.md %}) were introduced in v22.2. [#126413][#126413]

> - Fixed a bug where a hash sharded constraint could not be created if it referred to columns that had a backslash in the name. [#128676][#128676]

Reviewer: ```suggestion
- Fixed a bug where a hash-sharded constraint could not be created if it referred to columns that had a backslash in the name. [#128676][#128676]
```

- Fixed a bug where `TYPEDESC SCHEMA CHANGE` jobs could end up retrying forever if the descriptor targeted by them was already dropped. [#128461][#128461]

> - Fixed a bug in which the output of [`EXPLAIN (OPT, REDACT)`]({% link v23.2/explain.md %}) for various `CREATE` statements was not redacted. This bug had existed since [`EXPLAIN (REDACT)`]({% link v23.2/explain.md %}#parameters) was introduced in v23.1 and affects the following statements: `EXPLAIN (OPT, REDACT) CREATE TABLE`, `EXPLAIN (OPT, REDACT) CREATE VIEW`, `EXPLAIN (OPT, REDACT) CREATE FUNCTION`. [#128488][#128488]

Reviewer: ```suggestion
- Fixed a bug in which the output of [`EXPLAIN (OPT, REDACT)`]({% link v23.2/explain.md %}) for various `CREATE` statements was not redacted. This bug had existed since [`EXPLAIN (REDACT)`]({% link v23.2/explain.md %}#parameters) was introduced in v23.1 and affects the following statements:
  - `EXPLAIN (OPT, REDACT) CREATE TABLE`
  - `EXPLAIN (OPT, REDACT) CREATE VIEW`
  - `EXPLAIN (OPT, REDACT) CREATE FUNCTION` [#128488][#128488]
```
